### PR TITLE
remove SHGetKnownFolderPath

### PR DESCRIPTION
**Pull request by ramarag**

### CODE REVIEW

**Comment by ramarag** [comment]
@gkhanna79 @eerhardt PTAL
cc @MichaelSimons

File: src/corehost/common/pal.windows.cpp [hunk 1]
```diff
@@ -134,48 +134,50 @@ void pal::unload_library(dll_t library)
...
+    recv->assign(file_path);
     return true;
 }
 
 static
 bool get_wow_mode_program_files(pal::string_t* recv)
 {
 #if defined(_TARGET_AMD64_)
-    KNOWNFOLDERID kfid = FOLDERID_ProgramFilesX86;
+    pal::char_t* kfid = _X("ProgramFiles(x86)");
```
Comment (gkhanna79) [inline src/corehost/common/pal.windows.cpp]: Since this is no longer KNOWNFOLDERID, can you please update the variable name to reflect it?

Comment (gkhanna79) [inline src/corehost/common/pal.windows.cpp]: Also, can you please work with @MichaelSimons to confirm that this works, as expected, on Nano?

Comment (ramarag) [inline src/corehost/common/pal.windows.cpp]: i already checked on nano, it works as expected

File: src/corehost/common/pal.windows.cpp [hunk 2]
```diff
@@ -190,7 +192,8 @@ bool pal::get_default_servicing_directory(string_t* recv)
 
 bool pal::get_global_dotnet_dir(pal::string_t* dir)
 {
-    if (!get_program_files_by_id(FOLDERID_ProgramFiles, dir))
+
```
Comment (gkhanna79) [inline src/corehost/common/pal.windows.cpp]: NIT: Extra line

**Comment by ramarag** [comment]
@dotnet-bot test Ubuntu16.10 x64 Release Build

**Pull request merged**